array_api_strict/tests/test_searching_functions.py

```diff
@@ -42,3 +48,54 @@ def test_where_f32():
     res = xp.where(xp.asarray([True, False]), 1., xp.asarray([2, 2], dtype=xp.float32))
     assert res.dtype == xp.float32
 
+
+@pytest.mark.parametrize("device", ALL_DEVICES)
+def test_where_device_persists(device):
+    """Test that the device of the input and output array are the same"""
+
+    cond = xp.asarray([True, False], device=device)
+    x1 = xp.asarray([1, 2], device=device)
+    x2 = xp.asarray([3, 4], device=device)
+    res = xp.where(cond, x1, x2)
+    assert res.device == device
+    res = xp.where(cond, 1, x2)
+    assert res.device == device
+    res = xp.where(cond, x1, 2)
+    assert res.device == device
```

Comment:
This and the following tests are great. I imaging we'll want to parrot them in array-api-tests at some point.